go/oasis-test-runner/scenario/e2e/runtime/sentry.go

```diff
@@ -130,3 +157,246 @@ func (s *sentryImpl) Fixture() (*oasis.NetworkFixture, error) {
+	opts = &cmnGrpc.ClientOptions{
+		CommonName: identity.CommonName,
+		ServerPubKeys: map[signature.PublicKey]bool{
+			sentry0.GetTLSPubKey(): true,
+		},
+		Certificates: []tls.Certificate{*validator1Identity.TLSSentryClientCertificate},
+	}
+	conn, err = s.dial(sentry0Address, opts, false)
+	if err != nil {
+		return fmt.Errorf("sentry: dial error: %w", err)
+	}
+	defer conn.Close()
+	sentry0Client = api.NewSentryClient(conn)
+	_, err = sentry0Client.GetAddresses(ctx)
+	s.Logger.Debug("sentry0.GetAddress with validator1 sentry cert", "err", err)
+	if status.Code(err) != codes.PermissionDenied {
+		return errors.New("sentry0 control endpoint should deny connection with validator1 certificate")
+	}
+
+	// Check Sentry-0 control endpoint with Validator-0 client certificates.
+	opts = &cmnGrpc.ClientOptions{
+		CommonName: identity.CommonName,
+		ServerPubKeys: map[signature.PublicKey]bool{
+			sentry0.GetTLSPubKey(): true,
+		},
+		Certificates: []tls.Certificate{*validator0Identity.TLSSentryClientCertificate},
+	}
+	conn, err = s.dial(sentry0Address, opts, false)
+	if err != nil {
+		return fmt.Errorf("sentry: dial error: %w", err)
+	}
+	defer conn.Close()
+	sentry0Client = api.NewSentryClient(conn)
+	_, err = sentry0Client.GetAddresses(ctx)
+	s.Logger.Debug("sentry0.GetAddress with validator0 sentry cert", "err", err)
+	if err != nil {
+		return errors.New("sentry0 control endpoint should allow connection with validator0 certificate")
+	}
+
+	// Sanity check storage endpoints. Only committee and configured upstream sentry nodes
+	// are allowed to access corresponding storage node write endpoints.
+
+	// Check Storage-1 endpoint with Validator-0 client certificates.
+	opts = &cmnGrpc.ClientOptions{
+		CommonName:    identity.CommonName,
+		ServerPubKeys: map[signature.PublicKey]bool{},
+		Certificates:  []tls.Certificate{*validator0Identity.TLSSentryClientCertificate},
+	}
+	// XXX: we need to skip client authentication of the server keys as storage-1 is using ephemeral
+	// tls keys which aren't known.
+	conn, err = s.dial(storage1Address, opts, true)
+	if err != nil {
+		return fmt.Errorf("sentry: dial error: %w", err)
+	}
+	defer conn.Close()
+	storage1Client := storage.NewStorageClient(conn)
+	_, err = storage1Client.GetCheckpoints(ctx, emptyGetCheckpointsReq)
+	s.Logger.Debug("storage1.GetCheckpoints with validator0 cert", "err", err)
+	if status.Code(err) != codes.PermissionDenied {
+		return errors.New("storage1 checkpoint endpoint should deny connection with validator0 certificate")
+	}
+
+	// Check Storage-1 endpoint with Storage-0 client certificates.
+	opts = &cmnGrpc.ClientOptions{
+		CommonName:    identity.CommonName,
+		ServerPubKeys: map[signature.PublicKey]bool{},
+		Certificates:  []tls.Certificate{*storage0Identity.GetTLSCertificate()},
+	}
+	// XXX: we need to skip client authentication of the server keys as storage-1 is using ephemeral
+	// tls keys which aren't known.
```

Comment:
Actually, with the new `GetStatus` endpoint you can now fetch them by calling `GetStatus` on the target node and inspecting the retrieved `Identity` field (or even the node registration descriptor) in [the response](https://pkg.go.dev/github.com/oasisprotocol/oasis-core/go/control/api?tab=doc#Status).

Reply:
Registration descriptor wouldn't work as i believe its sentry node key is used there, but using the `Identity` field should work yeah. Nice wasn't aware of the endpoint, in that case i also don't need the node with cert rotation disabled - but will probably keep the "refactor" in (moving the flag from oasis.go into node configs).

Reply:
Fixed, thanks for the suggestion, no need for the hack with skipping client auth anymore.